### PR TITLE
Test time series id in RecoverySourceHandlerTests

### PR DESCRIPTION
**Pull request by nik9000**
We were just testing the standard `_id`. This converts
`RecoverySourceHandlerTests` into a `MapperServiceTestCase` so it can do
a "real" parse of the document which'll be very important given our
future plans to make `_id` non-stored in tsdb.

### CODE REVIEW

**Comment by nik9000** [comment]
run elasticsearch-ci/part-2

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java [hunk 1]
```diff
@@ -462,29 +462,72 @@ public void indexTranslogOperations(
...
-        );
-        return new Engine.Index(new Term("_id", Uid.encodeId(doc.id())), randomNonNegativeLong(), doc);
+    private interface TestCase {
+        Engine.Index createIndexOp(int docIdent);
+    }
+
+    private class StandardModeTestCase implements TestCase {
+        private final MapperService mapper;
+
+        private StandardModeTestCase() throws IOException {
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java:472]: I think you wanted to randomly choose between the two test-cases? This one is not in use now and I think it should be.

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java [hunk 2]
```diff
@@ -462,29 +462,72 @@ public void indexTranslogOperations(
...
-            seqID,
-            id,
-            null,
-            Arrays.asList(document),
-            source,
-            XContentType.JSON,
-            null
-        );
-        return new Engine.Index(new Term("_id", Uid.encodeId(doc.id())), randomNonNegativeLong(), doc);
+    private interface TestCase {
```
Comment (henningandersen) [inline server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java:465]: Can we call this `IndexOpFactory` or similar?

**Review by henningandersen — APPROVED** [review]
LGTM.

Left a couple of comments that I would like to see fixed, otherwise good.

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java (hunk 1 above)
Comment (nik9000) [reply server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java:472]: Boo. yeah. Sorry.

File: server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java (hunk 2 above)
Comment (nik9000) [reply server/src/test/java/org/elasticsearch/indices/recovery/RecoverySourceHandlerTests.java:465]: Sure!

**Pull request merged**